writeToStream.js

```diff
@@ -0,0 +1,579 @@
+
+'use strict';
+
+var protocol = require('./constants')
+  , empty = new Buffer(0)
+  , zeroBuf = new Buffer([0])
+  , numCache = require('./numbers')
+
+function generate(packet, stream) {
+
```

Comment:
I guess you have left cork/uncork calls out of writeStream because of backwards compatibility, right? Even though I'd like to have a short-hand alternative that implicitly does cork/uncork maybe something like,
```
function alternative_generate(packet, stream) {
    stream.cork();
    generate(packet, stream);
    process.nextTick(uncork, stream);
}
```